tests/functional/test_play.py

```diff
@@ -0,0 +1,190 @@
+    │       │ test     │ rose-stem │ Class     │returned 0     │
+    │set-up │ specific │ runner    │ container ├───────────────┤
+    │repo   │ fixture  │ fixture   │           │Test for output│
+    │fixture│ (set up  │           │ Only run  │strings "foo"  │
+    │       │ rose-stem│ (run      │ class     ├───────────────┤
+    │       │ command) │ rose-stem)│ fixture   │Test for output│
+    │       │          │           │ once      │strings "bar"  │
+    │       ├──────────┼─ ── ── ── ├───────────┼───────────────┤
+    │       │          │           │           │               │
+    │       │          │           │           ├───────────────┤
+    │       │          │           │           │               │
+    │       │          │           │           ├───────────────┤
+    │       │          │           │           │               │
+
+Debugging
+---------
+Because of the tasks being run in subprocesses debugging can be a little
+tricky. As a result there is a commented ``breakpoint`` in
+``rose_stem_run_template`` indicating a location where it might be useful
+to investigate failing tests.
+
+
+"""
+
+import os
+import pytest
+import re
+import shutil
+import subprocess
+
+from pathlib import Path
+from uuid import uuid4
+
+from cylc.flow.hostuserutil import get_host
+from cylc.flow.pathutil import get_workflow_run_dir
+
+
+HOST = get_host()
+
+
+@pytest.fixture(scope='module')
+def monkeymodule():
+    from _pytest.monkeypatch import MonkeyPatch
+    mpatch = MonkeyPatch()
+    yield mpatch
+    mpatch.undo()
+
+
+@pytest.fixture(scope='module')
+def fixture_provide_flow(tmp_path_factory):
+    """Provide a cylc workflow based on the contents of a folder which can
+    be either validated or installed.
+    """
+    src_flow_name = '13_cylc_play'
+    workflow_src = Path(__file__).parent / src_flow_name
+    test_flow_name = f'cylc-rose-test-{str(uuid4())[:8]}'
+    srcpath = (tmp_path_factory.getbasetemp() / test_flow_name)
+    flowpath = Path(get_workflow_run_dir(test_flow_name))
+    shutil.copytree(workflow_src, srcpath)
+    yield {
+        'test_flow_name': test_flow_name,
+        'flowpath': flowpath,
+        'srcpath': srcpath
+    }
+    shutil.rmtree(srcpath)
+    shutil.rmtree(flowpath)
+
+
+@pytest.fixture(scope='module')
+def fixture_install_flow(fixture_provide_flow, monkeymodule):
+    """Run ``cylc install``.
+
+    By running in a fixture with modular scope we
+    can run tests on different aspects of its output as separate tests.
+
+    If a test fails using ``pytest --pdb then``
+    ``fixture_install_flow['result'].stderr`` may help with debugging.
+    """
+    result = subprocess.run(
+        [
+            'cylc', 'install',
+            '--flow-name', fixture_provide_flow['test_flow_name'],
+            '-C', str(fixture_provide_flow['srcpath'])
+        ],
+        capture_output=True,
+        env=os.environ
+    )
+    yield {
+        'fixture_provide_flow': fixture_provide_flow,
```

Comment:
I think this would be easier to use if the `fixture_provide_flow` dict was expanded into this one
```suggestion
        **fixture_provide_flow,
```
(Obviously this change would need corresponding ones for the tests that use the fixture)

Reply:
Excellent Idea.